tests/net/socket/tcp/src/main.c

```diff
@@ -723,6 +723,45 @@ void test_open_close_immediately(void)
 	k_sleep(TCP_TEARDOWN_TIMEOUT);
 }
 
+void test_open_obstructed(void)
+{
+	/* Test if socket connect fails when there is not communication
+	 * possible.
+	 */
+	int count_after = 0;
+	struct sockaddr_in c_saddr;
+	struct sockaddr_in s_saddr;
+	int c_sock;
+	int rv;
+
+	prepare_sock_tcp_v4(CONFIG_NET_CONFIG_MY_IPV4_ADDR, ANY_PORT,
+			    &c_sock, &c_saddr);
+
+	s_saddr.sin_family = AF_INET;
+	s_saddr.sin_port = htons(SERVER_PORT);
+	rv = zsock_inet_pton(AF_INET, CONFIG_NET_CONFIG_MY_IPV4_ADDR, &s_saddr.sin_addr);
+	zassert_equal(rv, 1, "inet_pton failed");
+
+	loopback_set_packet_drop_ratio(1.0f);
+
+	zassert_not_equal(connect(c_sock, (struct sockaddr *)&s_saddr,
+				  sizeof(s_saddr)),
+			  0, "connect succeed");
+
+	zassert_equal(errno, ETIMEDOUT,
+		"connect should be timed out, got %i", errno);
+
+	test_close(c_sock);
+
+	k_sleep(TCP_TEARDOWN_TIMEOUT);
```

Comment:
I don't think we need teardown delay in this case, not connection has been established.

Reply:
Removed